src/Controller/AcademiesController.php

```diff
@@ -14,27 +18,135 @@ class AcademiesController extends AbstractController
      *
      * @return JsonResponse
      *
-     * @Route("/api/v1/academies", name="academies")
-     * @method({'GET'})
+     * @Route(
+     *     "/api/v1/academies",
+     *     name="academies",
+     *     methods={"GET"}
+     * )
      */
-    public function index()
+    public function getAcademies()
     {
         $academies = $this->getDoctrine()
             ->getRepository(Academy::class)
             ->findAll();
 
+        if ($academies === null) {
+            return new JsonResponse("We could not find any academies", Response::HTTP_NOT_FOUND);
+        }
+
         $academiesArray = array();
 
         foreach ($academies as $academy) {
             $academiesArray[] = array(
-                'academy_id' => $academy->getId(),
-                'academy_name' => $academy->getAcademyName(),
-                'academy_email' => $academy->getAcademyEmail(),
-                'academy_url' => $academy->getAcademyUrl(),
-                'academy_logo' => $academy->getAcademyLogo(),
+                'academy_id'          => $academy->getId(),
+                'academy_name'        => $academy->getAcademyName(),
+                'academy_email'       => $academy->getAcademyEmail(),
+                'academy_url'         => $academy->getAcademyUrl(),
+                'academy_logo'        => $academy->getAcademyLogo(),
+                'academy_description' => $academy->getAcademyDescription(),
             );
         }
 
-        return new JsonResponse($academiesArray);
+        return new JsonResponse(
+            $academiesArray,
+            JsonResponse::HTTP_OK
+        );
+    }
+
+    /**
+     * Returns academy selected by the id.
+     *
+     * @param $id
+     * @return JsonResponse
+     *
+     * @Route(
+     *     "/api/v1/academy/{id}",
+     *     name="academy",
+     *     methods={"GET"},
+     *     requirements={"id"="\d+"}
+     * )
+     */
+    public function getAcademy(int $id)
+    {
+        $academy = $this->getDoctrine()
+            ->getRepository(Academy::class)
+            ->find($id);
+
+        if ($academy === null) {
+            return new JsonResponse("There is no academy with id of: " . $id, Response::HTTP_NOT_FOUND);
+        }
+
+        $programsArray = [];
+
+        foreach ($academy->getPrograms() as $program) {
+                $programsArray[] = [
+                    'academy_program' => [
+                        'program_name'      => $program->getProgramName(),
+                        'program_price'     => $program->getProgramPrice(),
+                        'program_review'    => $this->getReviews(),
```

Comment:
This will return only this program reviews or all reviews from DB? Because as far as I'm looking at `getReviews` method it selects in general all reviews, or am I missing something?

Reply:
It returns reviews that belong to a certain program, not all reviews in general.

Reply:
Look closely.
Are you 100% sure that this code will work as you described?